src/linearAlgebra/BlasLapackKernels.h

```diff
@@ -90,50 +90,52 @@ namespace dftefe
         static std::vector<double>
         nrms2MultiVector(size_type               vecSize,
                          size_type               numVec,
-                         const ValueType *       multiVecData,
+                         const ValueType1 *      multiVecData,
                          BlasQueue<memorySpace> &BlasQueue);
       };
 
-
 #ifdef DFTEFE_WITH_DEVICE
-      template <typename ValueType>
-      class Kernels<ValueType, dftefe::utils::MemorySpace::DEVICE>
+      template <typename ValueType1, typename ValueType2>
```

Comment:
Shouldn't this be `template <typename ValueType1, typename ValueType2 = ValueType1>`?

Reply:
I tried that but C++ complains that default template argument is not allowed during partial template specialization of class template